docs/runtimes/node-affinity.md

```diff
@@ -1,96 +1,214 @@
-reboot, or terminate it. With spot instances you request EC2 capacity from specific availability zones and is 
+
+### On Demand vs. Spot 
+
+On-demand instances provide full control over the instance lifecycle. You decide when to launch, stop, hibernate, start, 
+reboot, or terminate it. With Spot instances you request capacity from specific availability zones, though it is  
 susceptible to spot capacity availability. This is a good choice if you can be flexible about when your applications run 
-and if your applications can be interrupted.
+and if your applications can be interrupted. 
+
+### Stateless and Stateful Applications 
+When deploying your MLRun jobs to specific nodes, take into consideration that On-demand 
+nodes are designed to run stateful applications while spot nodes are designed for stateless applications. 
+MLRun jobs are more stateful by nature. An MLRun job that is assigned to run on a spot node might be subject to interruption; 
+it would have to be designed so that the job/function state will be saved when scaling to zero.
+
+```{admonition} Important
+When an MLRun job is running on a spot node and it fails, it won't get back up again. Howewer, if Nuclio goes down due to a spot issue, it would be brought up by Kubernetes.
+```
+
+### UI configuration
+
+```{admonition} Note
+Relevant when MLRun is executed in the [Iguazio platform](https://www.iguazio.com/docs/latest-release/).
+```
 
-## Stateless and stateful applications 
-When deploying your MLRun jobs to specific nodes, please take into consideration that on demand 
-nodes are best designed to run stateful applications while spot nodes are best designed to stateless applications. 
-MLRun jobs which are stateful, and are assigned to run on spot nodes, may be subject to interruption 
-and will to be designed so that the job/function state will be saved when scaling to zero.
+You can configure Spot node support when creating a job, rerunning an existing job, and creating an ML function. 
+The **Run on Spot nodes** drop-down list is in the **Resources** section of jobs. 
+Configure the Spot node support for individual Nuclio functions when creating a function in the **Configuration** tab, under **Resources**. 
 
-## Node selector
-Using the **Node Selector** you can assign MLRun jobs to specific nodes within the cluster. 
-**Node Selector** is available for all modes of deployment in the platform including the platform UI, 
-command line, and programmable interfaces.
+### SDK configuration
 
-To assign MLRun jobs to specific nodes you use the Kubernetes node label 
-`app.iguazio.com/lifecycle` with the values of:
+Configure preemption mode by adding the parameter in your Jupyter notebook, specifying a mode from the list of values above. <br>
+For example:
 
-* preemptible – assign to EC2 Spot instances
+```func.with_preemption_mode(mode={value})```
```

Comment:
need more complete example (few lines), at least for the first example (e.g. create a function and set the mode)